hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/ProtocolMessageMetrics.java

```diff
@@ -40,32 +44,59 @@ public class ProtocolMessageMetrics<KEY> implements MetricsSource {
 
   private final String description;
 
+  private final MetricsRegistry registry;
+
+  private final boolean quantileEnable;
+
   private final Map<KEY, AtomicLong> counters =
       new ConcurrentHashMap<>();
 
   private final Map<KEY, AtomicLong> elapsedTimes =
       new ConcurrentHashMap<>();
 
+  private final Map<KEY, MutableQuantiles[]> quantiles =
+      new ConcurrentHashMap<>();
+
   private final AtomicInteger concurrency = new AtomicInteger(0);
 
   public static <KEY> ProtocolMessageMetrics<KEY> create(String name,
-      String description, KEY[] types) {
-    return new ProtocolMessageMetrics<KEY>(name, description, types);
+      String description, KEY[] types, ConfigurationSource conf) {
+    return new ProtocolMessageMetrics<KEY>(name, description, types, conf);
   }
 
   public ProtocolMessageMetrics(String name, String description,
-      KEY[] values) {
+      KEY[] values, ConfigurationSource conf) {
     this.name = name;
     this.description = description;
+    registry = new MetricsRegistry(name + "MessageMetrics");
+    int[] intervals = conf.getInts(
+        OzoneConfigKeys.OZONE_PROTOCOL_MESSAGE_METRICS_PERCENTILES_INTERVALS);
+    quantileEnable = (intervals.length > 0);
     for (KEY value : values) {
       counters.put(value, new AtomicLong(0));
       elapsedTimes.put(value, new AtomicLong(0));
+      if (quantileEnable) {
+        MutableQuantiles[] mutableQuantiles =
+            new MutableQuantiles[intervals.length];
+        quantiles.put(value, mutableQuantiles);
+        for (int i = 0; i < intervals.length; i++) {
+          mutableQuantiles[i] = registry.newQuantiles(
+              value.toString() + "RpcTime" + intervals[i] + "s",
+              value.toString() + "rpc time in milli second",
+              "ops", "latency", intervals[i]);
```

Comment:
Now the name of the Metric will be like InfoVolumeRpcTime127s90thPercentileLatency and the unit of time is including in it as "s". It will be appreciated if you could tell me how to let it more suitable. @xichen01

Reply:
I think that `InfoVolumeRpcTime127s90thPercentileLatencyMs` is available, the `127s` is the time of sampling, `Ms` is the unit of the Metrics value.

Reply:
> InfoVolumeRpcTime127latencySeconds90thPercentileLatency

Thank you for your feedback! I appreciate your clarification on the suggestion you provided. I now understand it better, and I will proceed to make the necessary modifications according to your advice.